### DISCUSSION TITLE
Support for balancing/equalizing panes?

### DISCUSSION THREAD

**Discussion by bcspragu** [zellij-org/zellij#3662] [Q&A]
Hello!

Does Zellij have support for making all panes the same size or otherwise equalizing them? I notice when I create a bunch of panes in one direction, they end up looking like this:

![2024-10-10_11-17-44](https://github.com/user-attachments/assets/69196229-95d2-4cff-bf44-ead3c1f2bac1)

And of course, I can manually resize these in pane mode, or create swap layouts, but sometimes I just want to rebalance my current layout, while keeping the relative positioning

I use this functionality in my window manager ([bspwm](https://github.com/baskerville/bspwm)), which supports both "equalizing" and "balancing", which it describes as:

```
-E, --equalize
   Reset the split ratios of the tree rooted at the selected node to their default value.
                                                                                                                  
-B, --balance
   Adjust the split ratios of the tree rooted at the selected node so that all windows occupy the same area.
```

I'd be up for contributing this if it doesn't exist and there's interest.

**Comment by imsnif** [comment]
IMO the best way to achieve something like this would be as you said through a swap layout. The cool thing about them is that you can "snap back" into them with your existing panes. So if you have a "balanced" swap layout and reach this stage, you can do `Alt ]` or `Alt [` to snap the current pane state into the next swap layout and have them balanced.

Also, if you have a "balanced" swap layout you can probably use `Alt n` instead of horizontal splitting them and they will become balanced.

Indeed, even if we were to include such a feature in core, the cleanest way to implement it would be to use a swap layout internally anyway. So I very much recommend implementing it that way for your use case.
